### DISCUSSION TITLE
Simple program is using too much resources

### DISCUSSION THREAD

**Discussion by coldspice** [gfx-rs/wgpu#4679] [Q&A]
I'm following this tutorial. when I run it its using 40% of GPU 

https://github.com/sotrh/learn-wgpu/blob/master/code/beginner/tutorial4-buffer/Cargo.toml

[dependencies]
winit = "0.28"
env_logger = "0.10"
log = "0.4"
wgpu = "0.17"

**Comment by waynr** [comment]
I took a quick peek at this and noticed that if I removed the following line the GPU usage goes down: https://github.com/sotrh/learn-wgpu/blob/677300a5e96fa0b06ee7c3b4e09eb9bd6fee2e88/code/beginner/tutorial4-buffer/src/lib.rs#L372-L376

But...it's still kind of high for such a simple program. I would like to better understand what causes GPU utilization to increase.

**Reply by waynr** [reply]
(to be clear, i believe my fix significantly reduces the update frequency of the framebuffer or whatever since now the only time it should update is when there is a window event that necessitates redraw.)
